src/utils/date-range.js

```diff
@@ -40,18 +49,21 @@ const parseTwoDates = date => {
         return `${m1} ${d1} – ${m2} ${d2}`;
       }
     } else {
-      if (inOneDay) {
-        // Jan 12, 2020
-        return `${m1} ${d1}, ${y1}`;
+      if (inOneMonth) {
+        if (inOneDay) {
+          // Jan 12, 2020
+          return `${m1} ${d1}, ${y1}`;
+        } else {
+          // Jun 12 - 15, 2020
+          return `${m1} ${d1} – ${d2}, ${y1}`;
+        }
       } else {
-        // Jun 12 - 15, 2020
-        return `${m1} ${d2} – ${d2}, ${y1}`;
+        // Jun 12 - Jul 15, 2020
+        return `${m1} ${d1} – ${m2} ${d2}, ${y1}`;
       }
```

Comment:
Check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat/formatRange

Reply:
Seems like the support is poor